### PR TITLE
Feat/incentives settlement

### PR DESCRIPTION
**Pull request by andytcf**

### CODE REVIEW

File: queries/liquidityPools/useCurvePoolQuery_1.ts [hunk 1]
```diff
@@ -144,6 +144,7 @@ const useCurvePoolQuery_1 = (options?: QueryConfig<CurveData>) => {
 				duration: Number(duration) * 1000,
 				allowance,
 				userBalance,
+				needsToSettle: false,
```
Comment (dvd-schwrtz) [inline queries/liquidityPools/useCurvePoolQuery_1.ts:147]: you dont need to have this on all of them but its ok if you do. you can always do `needsToSettle?: boolean` when you define it instead.

File: sections/earn/Incentives.tsx [hunk 2]
```diff
@@ -102,6 +103,7 @@ const Incentives: FC<IncentivesProps> = ({
 							now,
 							route: ROUTES.Earn.Curve_LP,
 							tab: Tab.Curve_LP,
+							needsToSettle: false,
```
Comment (dvd-schwrtz) [inline sections/earn/Incentives.tsx:106]: same comment as above for the props you pass in here. doesn't need this field for all entries

File: sections/earn/LPTab/Settle/Settle.tsx [hunk 3]
```diff
@@ -0,0 +1,281 @@
...
+		throw new Error('unrecognizable asset');
+	}
+};
+
+type ApproveProps = {
+	stakedAsset: CurrencyKey;
+	setShowSettleOverlayModal: (show: boolean) => void;
+};
+
+const Approve: FC<ApproveProps> = ({ stakedAsset, setShowSettleOverlayModal }) => {
```
Comment (dvd-schwrtz) [inline sections/earn/LPTab/Settle/Settle.tsx:72]: you didn't change the component name

**Review by dvd-schwrtz — APPROVED** [review]

File: sections/earn/LPTab/Settle/Settle.tsx [hunk 4]
```diff
@@ -0,0 +1,281 @@
...
+	const { contracts, utils } = synthetix.js!;
+	if (stakedAsset === Synths.iBTC) {
+		return {
+			contract: contracts.Exchanger,
+			synth: utils.formatBytes32String('iBTC'),
+		};
+	} else if (stakedAsset === Synths.iETH) {
+		return {
+			contract: contracts.Exchanger,
+			synth: utils.formatBytes32String('iETH'),
```
Comment (evgenyboxer) [inline sections/earn/LPTab/Settle/Settle.tsx:60]:
```suggestion
			synth: utils.formatBytes32String(Synths.iETH),
```

File: sections/earn/LPTab/Settle/Settle.tsx [hunk 5]
```diff
@@ -0,0 +1,281 @@
...
+} from '../../common';
+import Color from 'color';
+import { walletAddressState } from 'store/wallet';
+
+export const getSettleSynthType = (stakedAsset: CurrencyKey) => {
+	const { contracts, utils } = synthetix.js!;
+	if (stakedAsset === Synths.iBTC) {
+		return {
+			contract: contracts.Exchanger,
+			synth: utils.formatBytes32String('iBTC'),
```
Comment (evgenyboxer) [inline sections/earn/LPTab/Settle/Settle.tsx:55]:
```suggestion
			synth: utils.formatBytes32String(Synths.iBTC),
```

File: sections/earn/LPTab/Settle/Settle.tsx [hunk 6]
```diff
@@ -0,0 +1,281 @@
...
+						setTxModalOpen(false);
+					}
+				} catch (e) {
+					setTransactionState(Transaction.PRESUBMIT);
+					setError(e.message);
+				}
+			}
+		}
+		approve();
+	}, [stakedAsset, provider, gasPrice, monitorHash, isAppReady]);
```
Comment (evgenyboxer) [inline sections/earn/LPTab/Settle/Settle.tsx:146]:
```suggestion
	}, [stakedAsset, provider, gasPrice, monitorHash, isAppReady, walletAddress]);
```

File: sections/earn/LPTab/LPTab.tsx [hunk 7]
```diff
@@ -97,14 +108,20 @@ const LPTab: FC<LPTabProps> = ({ stakedAsset, tokenRewards, allowance, userBalan
 				key: 'unstake',
 			},
 		];
-	}, [t, stakedAsset, userBalance, staked]);
+	}, [t, stakedAsset, userBalance, staked, needsToSettle]);
```
Comment (evgenyboxer) [inline sections/earn/LPTab/LPTab.tsx:111]:
```suggestion
	}, [t, stakedAsset, userBalance, staked]);
```

File: queries/liquidityPools/useIBTCPoolQuery_1.ts [hunk 8]
```diff
@@ -33,23 +37,28 @@ const useIBTCPoolQuery_1 = (options?: QueryConfig<LiquidityPoolData>) => {
...
-				synthetix.js?.contracts.ProxyiBTC.allowance(walletAddress, address),
+				StakingRewardsiBTC.rewardRate(),
+				StakingRewardsiBTC.periodFinish(),
+				ProxyiBTC.balanceOf(address),
+				ProxyiBTC.balanceOf(walletAddress),
+				ExchangeRates.rateForCurrency(synthetix.js?.toBytes32(Synths.iBTC)),
+				StakingRewardsiBTC.earned(walletAddress),
+				StakingRewardsiBTC.balanceOf(walletAddress),
+				ProxyiBTC.allowance(walletAddress, address),
+				Exchanger.settlementOwing(walletAddress, formatBytes32String('iBTC')),
```
Comment (evgenyboxer) [inline queries/liquidityPools/useIBTCPoolQuery_1.ts:51]:
```suggestion
				Exchanger.settlementOwing(walletAddress, formatBytes32String(Synths.iBTC)),
```

**Review by evgenyboxer — APPROVED** [review]
@andytcf great work! left some minor changes

**Pull request merged**